mongodump/mongodump_test.go

```diff
@@ -579,9 +587,47 @@ func TestMongoDumpBSON(t *testing.T) {
 					md.OutputOptions.Archive = "dump_slash.archive"
 					err = md.Init()
 					So(err, ShouldBeNil)
+				})
+			})
+
+			Convey("and that it dumps a collection with a name >238 bytes in the right format", func() {
+				session, err := testutil.GetBareSession()
+				So(err, ShouldBeNil)
+				coll := session.Database(testDB).Collection(longCollectionName)
+				_, err = coll.InsertOne(nil, bson.M{"a": 1})
+				So(err, ShouldBeNil)
+				defer coll.Drop(nil)
+
+				md.ToolOptions.Namespace.Collection = longCollectionName
+				err = md.Init()
+				So(err, ShouldBeNil)
 
+				path, err := os.Getwd()
+				So(err, ShouldBeNil)
+
+				absDumpDir := util.ToUniversalPath(filepath.Join(path, "dump_slash"))
+				So(os.RemoveAll(absDumpDir), ShouldBeNil)
+				So(fileDirExists(absDumpDir), ShouldBeFalse)
+
+				dumpDBDir := util.ToUniversalPath(filepath.Join("dump_slash", testDB))
+				So(fileDirExists(dumpDBDir), ShouldBeFalse)
+
+				md.OutputOptions.Out = "dump_slash"
+				err = md.Dump()
```

Comment:
Unfortunately for this test I couldn't use `testDumpOneCollection()` as it's written, since it hinges on the fact that the bson/metadata file names are 1-to-1 with the collection they came from. It does this when it checks that the dumped bson data is correct.

I was thinking if it's possible to adjust it, but I'd probably end up copying all the new mongorestore code over. It doesn't seem worth it for solely checking that the file names are correct, so I do that manually a few lines down.

Reply:
I think it's fine doing the test here :+1: